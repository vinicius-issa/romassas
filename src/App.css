h1{
    font-family: Roboto, 'Open Sans', 'Helvetica Neue', sans-serif;
    text-align: center;
    color: #444;
}


.flex-container{
    display: flex !important;
    flex-direction: column;
    width: 100%;
}
