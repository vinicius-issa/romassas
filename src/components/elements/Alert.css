.alert{
    position: fixed;
    margin: -20px 0px;
    align-self: center;
    z-index: 10;
}
