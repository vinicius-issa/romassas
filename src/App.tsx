import React from 'react';
import { Container } from '@material-ui/core';

import Route from './routes'
import './App.css'
import Alert from './components/elements/Alert'



const App: React.FC<{}> = ()=> {
  return (
    <div className="App">
      <Container >
        <Alert />
        <Route/>
      </Container>
    </div>
  );
}


export default App;
