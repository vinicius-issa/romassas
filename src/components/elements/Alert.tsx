import React from 'react';
import { connect } from 'react-redux'
import { Container } from '@material-ui/core'
import { Alert } from '@material-ui/lab'

import { setErrorMessage } from '../../store/actions/errorMessage';
import './Alert.css'

interface Props {
    errorMessage: string;
    setError : any;
}

const AlertEl = ({errorMessage, setError}:Props) => {

    return (<Container id='alert-container' className='flex-container'>
                {errorMessage !== '' &&  
                    <Alert onClose={() => {setError('')}} className="alert" severity="error">{errorMessage}</Alert>
                }
            </Container>)
}

function mapStateToProps(state: any) {
    return {
        errorMessage: state.errorMessage.errorMessage
    }
  }
  
  function mapDispatchToProps(dispatch: any) {
    return {
        setError(error: string) {
            const action = setErrorMessage(error)
            dispatch(action);
        }
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
  )(AlertEl);
  