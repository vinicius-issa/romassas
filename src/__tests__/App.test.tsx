import React from 'react';
import App from '../App';
import Enzyme, {  mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Menu from '../components/elements/Menu';
import Alert from '../components/elements/Alert';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore();

const store = mockStore({errorMessage:{
  errorMessage: '',
}});

describe('Testing App Component', () => {
  it('should render App', () => {
    const wrapper = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );

    expect(wrapper.find(Menu)).toHaveLength(1);
    expect(wrapper.find(Alert)).toHaveLength(1);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
