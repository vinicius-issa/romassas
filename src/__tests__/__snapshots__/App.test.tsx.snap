// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component should render App 1`] = `"<div class=\\"App\\"><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"menu\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><h6 class=\\"MuiTypography-root header MuiTypography-h6\\">News</h6><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Login</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></header><div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\"><div class=\\"MuiContainer-root flex-container MuiContainer-maxWidthLg\\" id=\\"alert-container\\"></div><h1>Hello Wolrd</h1></div></div>"`;
